src/main/cs/RLBotDotNet/RLBotDotNet/Manager/BotManager.cs

```diff
@@ -115,15 +129,18 @@ private void MainBotLoop()
                 stopwatch.Restart();
 
                 // Set off events that end up running the bot code later down the line
-                botRunEvent.Set();
+                foreach (BotProcess proc in botProcesses)
+                {
+                    proc.botRunEvent.Set();
+                }
 
                 // Sleep efficiently (but inaccurately) for as long as we can
                 TimeSpan maxInaccurateSleepTime = targetSleepTime - stopwatch.Elapsed - timerResolution;
                 if (maxInaccurateSleepTime > TimeSpan.Zero)
                     Thread.Sleep(maxInaccurateSleepTime);
 
                 // We could sleep the rest of the time accurately with the use of a spin-wait, but since the main bot loop doesn't have to fire at precise intervals it's reasonable to omit this step
```

Comment:
This comment seems out of date now.

Does the spin wait have an adverse effect on overall CPU utilization, e.g. could it hurt the performance of Rocket League, twitch streaming, or other bots on the team? (in theory bots on the other team are shielded by CPU affinity)

Reply:
I found about 1% CPU usage in the diagnostics tool. I could add another parameter to the constructor that allows people to enable the spin wait. This will also help with backwards compatibility.